tests/playlists.py

```diff
@@ -96,6 +96,14 @@
 http://example.com/with-cc-low.m3u8
 '''
 
+VARIANT_PLAYLIST_WITH_VIDEO_CC_SUBS_AND_AUDIO = '''
+#EXTM3U
+#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=7680000,CLOSED-CAPTIONS="cc",SUBTITLES="sub",AUDIO="aud",VIDEO="vid"
+http://example.com/with-everything-hi.m3u8
+#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=65000,CLOSED-CAPTIONS="cc",SUBTITLES="sub",AUDIO="aud",VIDEO="vid"
+http://example.com/with-everything-low.m3u8
+'''
+
```

Comment:
I would use fake uri (i.e: `http://example.com/cc.srt`) instead of `"cc".

```
CLOSED-CAPTION does not specify a Rendition; the closed caption
      media is present in the Media Segments of every video Rendition.

      URI

      The value is a quoted-string containing a URI that identifies the
      Media Playlist file.  This attribute is OPTIONAL; see
      Section 4.3.4.2.1.  If the TYPE is CLOSED-CAPTIONS, the URI
      attribute MUST NOT be present.
```

Reply:
I think you're conflating the URI *attribute* of EXT-X-I-FRAME-STREAM-INF with the URI *line* that must follow EXT-X-STREAM-INF. In EXT-X-STREAM-INF, the CLOSED-CAPTIONS, AUDIO, VIDEO, and SUBTITLES attributes all reference a "GROUP-ID attribute of an EXT-X-MEDIA tag elsewhere".

However, reviewing this made me realize that StreamInfo for EXT-X-I-FRAME-STREAM-INF should have audio, subtitles, and closed-captions set to None explicitly.

> All attributes defined for the EXT-X-STREAM-INF tag (Section 4.3.4.2)
> are also defined for the EXT-X-I-FRAME-STREAM-INF tag, except for the
> FRAME-RATE, AUDIO, SUBTITLES, and CLOSED-CAPTIONS attributes.